Slim/Player/Protocols/HTTP.pm

```diff
@@ -286,9 +329,51 @@ sub canDirectStream {
 	return $url;
 }
 
+# TODO: what happens is this method is overloaded in a sub-class that does not call its parents method
```

Comment:
Should probably read "...what happens IF this method..."?

Do you plan to review the stock plugins implementing `canDirectStreamSong()`? I believe most of them would currently call `canDirectStream()` as this call does. Maybe some/many of them could be replaced with this (using the super class' implementation)?

Reply:
> Should probably read "...what happens IF this method..."?
> 
Yes
> Do you plan to review the stock plugins implementing `canDirectStreamSong()`? I believe most of them would currently call `canDirectStream()` as this call does. Maybe some/many of them could be replaced with this (using the super class' implementation)?
I will but I still need to figure out if this is a real issue or not. This evening, I was working on a solution to use the mp4 => aac parser even if ScanURL is not called.

Reply:
So I reviewed that properly. The initial call to check possibility of direct streaming is in Squeezebox2.pm and does
```
	if ($song && $handler && $handler->can("canDirectStreamSong")) {
		return $handler->canDirectStreamSong($client, $song);
	} elsif ($handler && $handler->can("canDirectStream")) {
		return $handler->canDirectStream($client, $url);
	}
```
But unfortunately, protocol handler like Deezer, Classical, WiMP, RemoteLibrary overload canDirectStreamSong with code like
```
sub canDirectStreamSong {
	my ( $class, $client, $song ) = @_;

	# We need to check with the base class (HTTP) to see if we
	# are synced or if the user has set mp3StreamingMethod
	return $class->SUPER::canDirectStream( $client, $song->streamUrl(), $class->getFormatForURL($song->track->url()) );
}
```
So they would have to be changed to benefit from the changes there and have can('canDirectStreamingSong') to be compatible with previous LMS version before trying the actual base class canDirectStreamigSong. Unless somebody has a better idea as I don't. I need the $song object to know if direct streaming can be done b/c the "stripheader" is set once the transcoding rules have been evaluated, and can only be done on $song class object, obviously not a $track which is not instantiated to a given player

Reply:
Man, am I behind.... I don't even know whether these comments about streaming plugins' protocol handler implementations are still valid:

> So they would have to be changed to benefit from the changes there and use can('canDirectStreamingSong') to be compatible with previous LMS version before trying the actual base class canDirectStreamigSong. Unless somebody has a better idea as I don't.
> I need the $song object to know if direct streaming can be done b/c the "stripheader" is set once the transcoding rules have been evaluated, and can only be done on $song class object, obviously not a $track which is not instantiated to a given player.

Can't you get the `$song` from the `$client` object?

```perl
my $song = $client->playingSong();
```

?

Reply:
Yes, these comments are still valid. The problem is that you might not get a $song (yet) on canStreamDirect, so you can't rely on it. 
```
	if ($song && $handler && $handler->can("canDirectStreamSong")) {
		return $handler->canDirectStreamSong($client, $song);
	} elsif ($handler && $handler->can("canDirectStream")) {
		return $handler->canDirectStream($client, $url);
	}
```

And you've not lost much as I made (too) many changes to finally converge to a proper solution. I'm a bit sorry to drag both of you back and forth like that.

Reply:
I'm just waiting until things settle down again and then when I find a chunk of time to review the changes.